app/controllers/data_files_controller.rb

```diff
@@ -169,6 +169,7 @@ def extract_samples
       SampleDataPersistJob.new(@data_file, @sample_type, User.current_user, assay_ids: params["assay_ids"]).queue_job
       flash[:notice] = 'Started creating extracted samples'
     else
+      @data_file.sample_persistence_task.update(status: "cancelled") if @data_file.sample_persistence_task&.success?
```

Comment:
Cancelling the persistence task did work (I think because it gets set to queued again before the line you pointed out), but destroying it is cleaner.
Destroying the sample_extraction_task causes an error at https://github.com/seek4science/seek/blob/main/app/models/task.rb#L37 and leaving it in doesn't seem to cause any issues, so I think its easier to just keep it.